### ISSUE TITLE
Graphic widget with more options

### ISSUE THREAD

**Issue by xescugc** [ebitenui/ebitenui#168]
https://github.com/user-attachments/assets/bac5e260-5444-426d-8889-f096426364e7

I was playing with the `widget.NewGraphic` to be able to have a button with an image, and then I needed that to behave with a `widget.Disabled = true` but it does not. So I was thinking we could add what the buttons have for the images, that we have different ones for different states.

```golang
widget.GraphicOpts.Images(widget.GraphicImage{
  Idle: ...,
  Disabled: ...,
})
```

Also the dotted line around it it's also a `Graphic`, it's actually 4, one for each one of the frames of the animation. That made me think that we could actually also have an option to play this type of static animations. Right now I'm just Hide/Show the right frame, but could be configured like:

```golang
widget.GraphicOpts.Animate(widget.GraphicAnimation{
  Images: []ebiten.Image{all the animations in order},
  Speed: // Number of Updates that have to pass in order to print the next animation,
})
```

Basically internally would have a count and do the typical `(count / speed) % len(images)` (`Speed` is a TBD name though haha)

I'm basically asking if any of this makes sense to be inside of the widget+lib of they have to be done outside of it as I'm doing now.

**Comment by mcarpenter622** [comment]
So, to answer your question, I do like the idea of animated graphics widgets. I think this is a capability that would be useful for ebitenui. Currently that Graphic widget is not super useful.
